scripts/gitlab/test-cpp.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+echo "________Running test-cpp.sh________"
+set -e # fail on any error
+set -u # treat unset variables as error
+#use nproc `linux only
+THREADS=$(nproc)
```

Comment:
are you aware that this yields 128 on ci2?

Reply:
is it bad?

Reply:
I'm not sure, but in general I think that overcommiting nodes hurts performance. this would be the case when submitting several jobs. Also I think this can simply be tested before merging into parity-ethereum master branch.

Reply:
this will be a subject to test later

Reply:
I don't quite understand. You make the changes now and test them later?

Reply:
it shoudn't influence other pipelines much, but if it will, I'll change the param to something like `nproc/4`